docs/process.md

```diff
@@ -0,0 +1,41 @@
+# Development Process
+
+Notes:
+* The development process is a simplified version of the one used in [TEAMMATES](https://github.com/TEAMMATES/repo/blob/master/devdocs/process.md). 
+* The `master` branch always contains the latest stable code (may contain unreleased features/fixes).
+* The `release` branch always contains the latest released version.
```

Comment:
How would our workflow change around this new branch? Do we merge into this instead of master?